docs/src/explanations/Documentation-of-functions.md

```diff
@@ -0,0 +1,316 @@
+  val out = DecoupledIO(UInt(8.W))
+  })
+  val p = Module(new PipelineStage)
+  val c = Module(new PipelineStage) 
+  // connect Producer to IO
+  p.io.a <> io.in
+  // connect producer to consumer
+  c.io.b <> p.io.b
+  // connect consumer to IO
+  io.out <> c.io.b
+}
+class PipelineStage extends Module{
+  val io = IO(new Bundle{
+    val a = Flipped(DecoupledIO(UInt(8.W)))
+    val b = DecoupledIO(UInt(8.W))
+  })
+  io.b <> io.a
+}
+```
+Below we can see the resulting error message for this example:
+```scala modc:crash
+ChiselStage.emitVerilog(new Wrapper)
+```
+### Conclusion:
+The `:=` operator goes field-by-field and attempts to connect the RHS to the LHS. If something on the LHS is actually an `Input`, or something on the RHS is an `Output`, you will get this error:
+
+
+## Concept 3: Always Use := to assign DontCare to Wires
+
+When assigning `DontCare` to something that is not directioned, should you use `:=` or `<>`? 
+We will find out using the sample codes below:
+( Scastie link for the experiment:https://scastie.scala-lang.org/Shorla/ZIGsWcylRqKJhZCkKWlSIA/1)
+
+```scala mdoc:silent:reset
+import chisel3._
+import chisel3.util.DecoupledIO
+
+class Wrapper extends Module{
+  val io = IO(new Bundle {
+  val in = Flipped(DecoupledIO(UInt(8.W)))
+  val out = DecoupledIO(UInt(8.W))
+  })
+  val p = Module(new PipelineStage)
+  val c = Module(new PipelineStage) 
+  //connect Producer to IO
+  io.in := DontCare
+  p.io.a <> DontCare
+  val tmp = Wire(Flipped(DecoupledIO(UInt(8.W))))
+  tmp := DontCare
+  p.io.a <> io.in
+  // connect producer to consumer
+  c.io.a <> p.io.b
+  //connect consumer to IO
+  io.out <> c.io.b
+}
+class PipelineStage extends Module{
+  val io = IO(new Bundle{
+    val a = Flipped(DecoupledIO(UInt(8.W)))
+    val b = DecoupledIO(UInt(8.W))
+  })
+  io.b <> io.a
+}
+```
+Below we can see the resulting Verilog for this example:
+```scala modc:crash
+ChiselStage.emitVerilog(new Wrapper)
+```
+### Conclusion:
+When `<>` was used to assign the unidrectioned wire `tmp` to DontCare, we got an error.
+But when `:=` was used to assign the wire to DontCare, no errors will occur.
+
+Thus, when assigning DontCare to a wire, always use `:=`.
+
+
+##  Concept 4: You can use <> or := to assign DontCare to directioned things (IOs)
+When assigning `DontCare` to something that is directioned, should you use `:=` or `<>`? 
+We will find out using the sample codes below:
+( Scastie link for the ecperiment:https://scastie.scala-lang.org/Shorla/ZIGsWcylRqKJhZCkKWlSIA/1)
+
+```scala mdoc:silent:reset
+import chisel3._
+import chisel3.util.DecoupledIO
+
+class Wrapper extends Module{
+  val io = IO(new Bundle {
+  val in = Flipped(DecoupledIO(UInt(8.W)))
+  val out = DecoupledIO(UInt(8.W))
+  })
+  val p = Module(new PipelineStage)
+  val c = Module(new PipelineStage) 
+  //connect Producer to IO
+  io.in := DontCare
+  p.io.a <> DontCare
+  val tmp = Wire(Flipped(DecoupledIO(UInt(8.W))))
+  tmp <> DontCare
```

Comment:
I don't think you need more in this example than these lines.

Reply:
Noted